Add controllers

Uzbagoissa opened the pull request.

Uzbagoissa committed:
Version 0.1 Work with User: add classes, controllers, errorhandlers

Uzbagoissa committed:
Version 0.1 Work with User: add classes, controllers, errorhandlers

Uzbagoissa committed:
Version 0.1
Work with User: add classes, controllers, errorhandlers

Uzbagoissa committed:
Version 0.2
Finish work with User: add blackListID for PostmanTests. usersTest - OK

Uzbagoissa committed:
Version 0.3
Work with Item in progress: Item create PostmanTests - OK

Uzbagoissa committed:
Version 1.0
Finish with User, Item. Add ForbiddenExeption, IncorrectParameterExeption. User Mapper, Item Mapper
PostmanTests - OK

Uzbagoissa committed:
Version 1.0
Make Checkstyle

jeggga commented on src/main/java/ru/practicum/shareit/booking/Booking.java:
```diff
@@ -1,7 +1,15 @@
+
+import java.util.Date;
+
 public class Booking {
```
Надо еще добавить аннотации для генерации конструкторов, геттеров, сетторе и тд

jeggga commented on src/main/java/ru/practicum/shareit/booking/BookingEnum.java:
```diff
@@ -0,0 +1,8 @@
+package ru.practicum.shareit.booking;
+
+public enum BookingEnum {
```
Не совсем понятное название. Вместо перечисления лучше написать статус

jeggga commented on src/main/java/ru/practicum/shareit/booking/dto/BookingDto.java:
```diff
@@ -1,7 +1,15 @@
+
+import java.util.Date;
+
 public class BookingDto {
```
Так же надо добавить аннотации, как и в классе `Booking`

jeggga commented on src/main/java/ru/practicum/shareit/item/model/Item.java:
```diff
@@ -1,7 +1,24 @@
+@Builder
 public class Item {
+    private long id;
+    private long userId;
```
Лучше прям юзера здесь хранить

Uzbagoissa replied:
К нам же приходит id юзера в ItemController, придется по дороге дергать UserService, чтобы сохранить User в Item. Не будет ли это лишним усложнением кода?

jeggga commented on src/main/java/ru/practicum/shareit/item/model/Item.java:
```diff
@@ -1,7 +1,24 @@
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
 public class Item {
```
Не увидел поля `request`

jeggga commented on src/main/java/ru/practicum/shareit/item/dto/ItemDto.java:
```diff
@@ -1,7 +1,24 @@
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
 public class ItemDto {
```
Так же нет реквеста

jeggga commented on src/main/java/ru/practicum/shareit/item/mapper/ItemMapper.java:
```diff
@@ -0,0 +1,31 @@
+
+@Service
+@RequiredArgsConstructor
+public class ItemMapper {
```
Можно маппер сделать статикой, только еще надо повесить ограничение на создание объекта. То есть приватный конструктор сделать

jeggga commented on src/main/java/ru/practicum/shareit/item/ItemRepositoryImpl.java:
```diff
@@ -0,0 +1,116 @@
+    @Override
+    public List<ItemDto> searchItems(String text) {
+        List<ItemDto> itemDtosSearched = new ArrayList<>();
+        if (text.equals("")) {
```
А чем не подошли методы `isEmpty или isBlank`?

Uzbagoissa replied:
Да, забыл про них

jeggga commented on src/main/java/ru/practicum/shareit/request/ItemRequest.java:
```diff
@@ -1,7 +1,12 @@
+
+import java.util.Date;
+
 public class ItemRequest {
```
Надо дописать аннотации

jeggga commented on src/main/java/ru/practicum/shareit/request/dto/ItemRequestDto.java:
```diff
@@ -1,7 +1,12 @@
+
+import java.util.Date;
+
 public class ItemRequestDto {
```
Аналогично

jeggga commented on src/main/java/ru/practicum/shareit/user/mapper/UserMapper.java:
```diff
@@ -0,0 +1,28 @@
+
+@Service
+@RequiredArgsConstructor
+public class UserMapper {
```
Такой же комментарий, как и к мапперу к модели `Item`

Uzbagoissa committed:
Version 1.1 Patch
Add Booking & ItemRequest mappers. Work with Booking & ItemRequest classes. Add field "request" in Item. Make mappers static.

Uzbagoissa committed:
Version 1.1 Patch
Checkstyle
